core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go

```diff
@@ -238,9 +243,28 @@ func (c *coordinator) ReportIsOnchain(ctx context.Context, epoch uint32, round u
 		return false, errors.Wrap(err, "log poller IndexedLogs")
 	}
 
-	c.lggr.Info(fmt.Sprintf("NewTransmission logs: %+v", logs))
+	// Filter for valid logs that match the current config digest.
+	var logsWithCorrectConfigDigest []logpoller.Log
+	for i := 0; i < len(logs); i++ {
+		if logs[i].EventSig == c.newTransmissionTopic {
```

Comment:
is this if check needed? seems like LP should always return events with `c.newTransmissionTopic`. If it doesn't, parsing will fail in line 251 and we will get an error.